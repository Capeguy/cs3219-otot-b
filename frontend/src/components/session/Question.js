import React from 'react'

import { CAccordion, CAccordionBody, CAccordionHeader, CAccordionItem, CCol } from '@coreui/react'
import PropTypes from 'prop-types'

const Question = ({ question }) => {
  return (
    <CCol>
      <h3>{question.title}</h3>
      <br />
      <p className="text-medium-emphasis" style={{ whiteSpace: 'preWrap' }}>
        {question.body}
      </p>
      <CAccordion alwaysOpen className="mt-3">
        {question.hints.map((hint, index) => (
          <div key={index}>
            <CAccordionItem itemKey={index}>
              <CAccordionHeader>Show Hint #{index + 1}</CAccordionHeader>
              <CAccordionBody>{hint}</CAccordionBody>
            </CAccordionItem>
          </div>
        ))}
      </CAccordion>
    </CCol>
  )
}

Question.propTypes = {
  question: PropTypes.object,
}

export default Question
