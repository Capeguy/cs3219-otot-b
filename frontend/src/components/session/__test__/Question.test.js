import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer';
import Question from 'src/components/session/Question'

const mockQuestion = {
    title: 'Question title',
    body: 'Question body',
    hints: [
        'hint number 1',
        'hint number 2',
        'hint number 3',
    ], 
}

it("renders without crashing", () => {
    shallow(<Question question={mockQuestion} />);
});

it('renders correctly', () => {
    const component = renderer.create(
        <Question question={mockQuestion} />,
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
