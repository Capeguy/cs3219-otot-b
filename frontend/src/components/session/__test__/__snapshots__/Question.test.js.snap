// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col"
>
  <h3>
    Question title
  </h3>
  <br />
  <p
    className="text-medium-emphasis"
    style={
      Object {
        "whiteSpace": "preWrap",
      }
    }
  >
    Question body
  </p>
  <div
    className="accordion mt-3"
  >
    <div>
      <div
        className="accordion-item"
      >
        <div
          className="accordion-header"
        >
          <button
            aria-expanded={true}
            className="accordion-button collapsed"
            onClick={[Function]}
          >
            Show Hint #
            1
          </button>
        </div>
        <div
          className="accordion-collpase collapse"
          style={
            Object {
              "height": undefined,
            }
          }
        >
          <div
            className="accordion-body"
          >
            hint number 1
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="accordion-item"
      >
        <div
          className="accordion-header"
        >
          <button
            aria-expanded={true}
            className="accordion-button collapsed"
            onClick={[Function]}
          >
            Show Hint #
            2
          </button>
        </div>
        <div
          className="accordion-collpase collapse"
          style={
            Object {
              "height": undefined,
            }
          }
        >
          <div
            className="accordion-body"
          >
            hint number 2
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="accordion-item"
      >
        <div
          className="accordion-header"
        >
          <button
            aria-expanded={true}
            className="accordion-button collapsed"
            onClick={[Function]}
          >
            Show Hint #
            3
          </button>
        </div>
        <div
          className="accordion-collpase collapse"
          style={
            Object {
              "height": undefined,
            }
          }
        >
          <div
            className="accordion-body"
          >
            hint number 3
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
